pyproject.toml

```diff
@@ -21,7 +21,6 @@ packages = [
 [tool.poetry.dependencies]
 python = "^3.6"
 diffsync = "^1.3.0"
-nautobot = ">=1.0.0b3,<2.0"
```

Comment:
Why remove this, just curious?

Reply:
Thanks for asking! I found that if I have `nautobot` specified as a dependency in `pyproject.toml`, then `poetry.lock` will lock in a specific version of `nautobot`, and when I do a `poetry install` into the Docker container, regardless of what version of Nautobot is pre-installed into the container, it will be overwritten by the version locked in `poetry.lock`, which is not desirable, to say the least. 

At first I just went with doing a `poetry remove nautobot` before `poetry install`, but then I realized this is not desirable in general - if a hypothetical future user has `nautobot 2.0` installed and does a `pip install nautobot-netbox-importer`, the last thing we would want to do is have pip automatically downgrade the installed version of nautobot to 1.99 or whatever. So removing this explicit dependency is actually IMHO the right thing to do - if you install the plugin with an incompatible version of Nautobot, nautobot has logic to refuse to load the plugin at runtime, and we should be simply relying on that instead.

https://github.com/python-poetry/poetry/discussions/3757 is loosely related to this issue.

Reply:
we experienced something similar in other development and did the same thing.  The drawback here in my mind is if you do a poetry install locally in your environment you won't get Nautobot now.  This drives all development to be in the container.  I'm not sure how I feel about that.

Reply:
True - for local development you'd need to manually install Nautobot from PyPI.